adafruit_requests.py

```diff
@@ -317,11 +317,16 @@ def _throw_away(self, nbytes: int) -> None:
         nbytes -= self._read_from_buffer(nbytes=nbytes)
 
         buf = self._receive_buffer
-        for _ in range(nbytes // len(buf)):
-            self._recv_into(buf)
-        remaining = nbytes % len(buf)
+        len_buf = len(buf)
+        for _ in range(nbytes // len_buf):
+            read = 0
+            while read < len_buf:
+                read += self._recv_into(buf, len_buf - read)
```

Comment:
How about this, which I think is a little clearer, and does less arithmetic:
```suggestion
            to_read = len_buf
            while to_read > 0:
                to_read -= self._recv_into(buf, to_read)
```

Reply:
I like it.